src/app/src/components/ModalSections/FileUpload.js

```diff
@@ -0,0 +1,222 @@
+import { useEffect, useState } from 'react';
+import {
+    Box,
+    Button,
+    Flex,
+    Heading,
+    Icon,
+    List,
+    ListItem,
+    Progress,
+    Text,
+} from '@chakra-ui/react';
+import { CloudUploadIcon } from '@heroicons/react/outline';
+
+import { convertIndexedObjectToArray } from '../../util/convertIndexedObjectToArray';
+import ModalSection from './ModalSection';
+
+export default function FileUpload({ PreviousButton, ContinueButton }) {
+    const [files, setFiles] = useState([]);
+
+    const addFiles = newFiles => setFiles(files => [...files, ...newFiles]);
+
+    return (
+        <ModalSection
+            preHeading='Optional'
+            heading='Would you like to add your current map?'
+            prevButton={PreviousButton}
+            nextButton={ContinueButton}
+        >
+            <Text>
+                If you would like to look at a reference or start from an
+                existing map you can upload it here. If you have a Shapefile, we
+                recommend adding it.
+            </Text>
+
+            <Flex mt={4} w='100%' grow>
+                <UploadBox setFiles={addFiles} />
+                <FilesBox files={files}></FilesBox>
+            </Flex>
+        </ModalSection>
+    );
+}
+
+function UploadBox({ setFiles }) {
+    const { hovering, handleUpload, startDrag, endDrag } =
+        useFileUpload(setFiles);
+
+    const openFileDialog = useFilePicker(setFiles);
+
+    const onLeaveDragBox = event => {
+        const enteredElement = event.relatedTarget;
+        if (elementIsNotInDragBox(enteredElement)) {
+            endDrag();
+        }
+    };
+
+    const elementIsNotInDragBox = element =>
+        !document.getElementById('drop_zone').contains(element);
+
+    return (
+        <Box
+            id='drop_zone'
+            w='50%'
+            onDrop={handleUpload}
+            onDragEnter={startDrag}
+            onDragLeave={onLeaveDragBox}
+            onDragEnd={endDrag}
+            onDragOver={event => event.preventDefault()}
+            border='1px dashed'
+            borderColor={hovering ? 'gray.900' : 'gray.200'}
+            bg={hovering ? 'rgba(0, 0, 0, 0.1)' : undefined}
+        >
+            <Flex
+                p={8}
+                direction='column'
+                textAlign='center'
+                align='center'
+                justify='center'
+            >
+                <CloudIconWithBackground />
+                <Text mb={2} fontFamily="'Inter', san-serif">
+                    Drag &amp; drop files here or
+                </Text>
+                <Button
+                    mb={6}
+                    color='gray.600'
+                    bg='gray.200'
```

Comment:
Let's use a variant here too, from the wireframes:

```suggestion
                    variant='primary'
```

```diff
diff --git a/src/app/src/theme.js b/src/app/src/theme.js
index 5a5af18..7b33e14 100644
--- a/src/app/src/theme.js
+++ b/src/app/src/theme.js
@@ -18,6 +18,19 @@ const Button = {
                 bg: 'gray.400',
             },
         },
+        primary: {
+            bg: 'gray.100',
+            color: 'gray.600',
+            _hover: {
+                bg: 'gray.600',
+                textDecoration: 'underline',
+                color: 'white',
+            },
+            _disabled: {
+                bg: 'gray.400',
+                color: 'white',
+            },
+        },
         secondary: {
             bg: 'white',
             borderWidth: '1px',
```

Reply:
5e26aac70909e3b39a6f05a2af87c0487d249eec